services/ui-src/src/components/fields/DateField.tsx

```diff
@@ -1,11 +1,11 @@
-import { useEffect, useState } from "react";
+import { useState } from "react";
```

Comment:
we can avoid using `useEffect` here if we do a direct check in the `onChangeHandler` before setting any form values.